From 40c0419ef10a0dcb592019a04556d56a96890520 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Wed, 5 Jun 2024 10:50:43 +0200
Subject: [PATCH] ARM: dts: stm32: Replace OOB interrupt by IB interrupt for
 WIFI on stm32mp157c-dk2

Enabling SDIO IRQ signalling on this interface instead of
out-of-band interrupt from brcmfmac

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Signed-off-by: Erwan SZYMANSKI <erwan.szymanski@st.com>
Change-Id: I3a016fa7471860401761c44e47e3c610e5c18714
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/385359
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp157c-dk2.dts | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/arch/arm/boot/dts/st/stm32mp157c-dk2.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2.dts
@@ -118,6 +118,7 @@
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	non-removable;
+	cap-sdio-irq;
 	st,neg-edge;
 	bus-width = <4>;
 	vmmc-supply = <&v3v3>;
@@ -129,9 +130,6 @@
 	brcmf: bcrmf@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
-		interrupt-parent = <&gpiod>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>; /* WL_HOST_WAKE */
-		interrupt-names = "host-wake";
 	};
 };
 
